### ISSUE TITLE
IE11 user login email validation fails if field has leading or trailing space

### ISSUE THREAD

**Issue by dnadle** [magento/magento2#6058]
### Preconditions

1. Magento 2.1.0
2. IE11 browser
3. Chrome 51 browser
### Steps to reproduce

1. Attempt login using IE with a valid user email but place a space either before or after the address.
2. Try again with Chrome.
### Expected result

1. Client-side email validation should work the same in IE11 as it does in Chrome. 
### Actual result

1. Using IE, client-side validation fails with "Please enter a valid email address (Ex: johndoe@domain.com)."
2. Using Chrome, login succeeds. (padded whitespace is presumably trimmed?)

**Comment by dnadle** [comment]
I received a report that Firefox behaves like IE11.

**Comment by magentodevleper** [comment]
Magento 2.1.0 also having same problem

**Comment by AVoskoboinikov** [comment]
Hi guys

Thanks for reporting this issue. I have created internal ticket MAGETWO-57657 to fix it

**Comment by magento-engcom-team** [comment]
@dnadle, thank you for your report.
We've created internal ticket(s) MAGETWO-57657 to track progress on the issue.

**Comment by mtbottens** [comment]
This is caused by browser inconsistencies in `element.value`. Per the spec, email inputs should trim leading and trailing whitespace. Chrome is doing this properly while ie11 (surprise) is not.

See spec here: https://www.w3.org/TR/html5/forms.html#e-mail-state-(type=email)

Code example:
```javascript
(function(document) {
    var input = document.createElement('input');
    input.type = 'email';
    input.value = ' test@example.com';
    console.log(input.value.indexOf(' ')); // -1 in chrome, 0 in IE
})(document)
```

**Comment by orlangur** [comment]
@mtbottens as I understand we cannot fix validation introduced by browser itself?

Such browser-specific validation could be disabled but I don't think it's a good idea.

**Comment by mtbottens** [comment]
@orlangur Essentially. No matter what we do, the browser is still going to return the value with the leading/trailing white space trimmed. The only thing we could really do is write a method to get the value which would handle trimming the whitespace for email inputs to support browsers which aren't following the spec. But doing that would require us overwriting logic in `$.fn.validate`, specifically the elementValue method.

I personally believe this should just be filed away as progressive enhancement and nothing should be done about it, but I suppose that is a decision left to Magento.

**Comment by dnadle** [comment]
You could add a validation for whitespace prior to the one for invalid email, so the validation error is meaningful and the user understands what is wrong.

**Comment by dankhrapiyush** [comment]
I am working on this at #dmcdindia

**Comment by magento-engcom-team** [comment]
@dankhrapiyush thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by magento-engcom-team** [comment]
Hi @dnadle. Thank you for your report. 
The issue has been fixed in magento/magento2#15365 by @dankhrapiyush in 2.2-develop branch
Related commit(s):
 - [9ecb84b8c17413eac3b125f3893644c161cc4e14](https://github.com/magento/magento2/commit/9ecb84b8c17413eac3b125f3893644c161cc4e14)
 - [62c1a146dab6f7d6c0e553de581e9886f9fedf44](https://github.com/magento/magento2/commit/62c1a146dab6f7d6c0e553de581e9886f9fedf44)
 - [c81d9a69d890990cb81a160036762c2dc1b44369](https://github.com/magento/magento2/commit/c81d9a69d890990cb81a160036762c2dc1b44369)

The fix will be available with the upcoming 2.2.6 release.

**Comment by sidolov** [comment]
Hi @dnadle. Thank you for your report. 
The issue has been fixed in magento/magento2#15874 by @dankhrapiyush in 2.1-develop branch
Related commit(s):
 - [f8397ebe12d6affa44ceeff40abad31d99678767](https://github.com/magento/magento2/commit/f8397ebe12d6affa44ceeff40abad31d99678767)
 - [43835fc4ae4cb1322d90bda02ba2726359171cf6](https://github.com/magento/magento2/commit/43835fc4ae4cb1322d90bda02ba2726359171cf6)
 - [14721c7d9bdb228988f8edc9efdaba9f0c241f8b](https://github.com/magento/magento2/commit/14721c7d9bdb228988f8edc9efdaba9f0c241f8b)

The fix will be available with the upcoming 2.1.15 release.

**Comment by magento-engcom-team** [comment]
Hi @dnadle. Thank you for your report. 
The issue has been fixed in magento/magento2#15884 by @dankhrapiyush in 2.3-develop branch
Related commit(s):
 - [c44f1f84d274ff9324742831e09f0b7f49027a7b](https://github.com/magento/magento2/commit/c44f1f84d274ff9324742831e09f0b7f49027a7b)

The fix will be available with the upcoming 2.3.0 release.

**Comment by VladimirZaets** [comment]
Hi @dnadle. Thank you for your report. 
The issue has been fixed in magento/magento2#16192 by @dankhrapiyush in 2.2-develop branch
Related commit(s):
 - [3cf1d74ee38f31a053963a797033cfa73290d9e8](https://github.com/magento/magento2/commit/3cf1d74ee38f31a053963a797033cfa73290d9e8)
 - [bd7d3b4bedf9db467018d068bf8d71869dadef33](https://github.com/magento/magento2/commit/bd7d3b4bedf9db467018d068bf8d71869dadef33)
 - [8a17d2c4ef07beed652807c654e91d135589b664](https://github.com/magento/magento2/commit/8a17d2c4ef07beed652807c654e91d135589b664)
- [a8948ecc61f9bbbfd19c84fb1c06767ca12ad9fa](https://github.com/magento/magento2/commit/a8948ecc61f9bbbfd19c84fb1c06767ca12ad9fa)

The fix will be available with the upcoming 2.2.6 release.

**Comment by sidolov** [comment]
Hi @dnadle. Thank you for your report. 
The issue has been fixed in magento/magento2#16297 by @dankhrapiyush in 2.1-develop branch
Related commit(s):
 - [96af9cfe69ee2d0ecb558e5205621e2cc4902f01](https://github.com/magento/magento2/commit/96af9cfe69ee2d0ecb558e5205621e2cc4902f01)
 - [5e58bacfe85e0af80620417c27d8dbd2294319bd](https://github.com/magento/magento2/commit/5e58bacfe85e0af80620417c27d8dbd2294319bd)
 - [4ec83ec9aac74ef120d15a3f95c3868138eda3c9](https://github.com/magento/magento2/commit/4ec83ec9aac74ef120d15a3f95c3868138eda3c9)

The fix will be available with the upcoming 2.1.15 release.

**Comment by sidolov** [comment]
Hi @dnadle. Thank you for your report. 
The issue has been fixed in magento/magento2#16293 by @dankhrapiyush in 2.3-develop branch
Related commit(s):
 - [89ca28f7ef9bde89d0a903513123db37e506dd1e](https://github.com/magento/magento2/commit/89ca28f7ef9bde89d0a903513123db37e506dd1e)
 - [5c1e8f770bca02fd3a3db78ce96def3843cfc50e](https://github.com/magento/magento2/commit/5c1e8f770bca02fd3a3db78ce96def3843cfc50e)
 - [bfce0265e495771e90350d206dbb0bf21132e879](https://github.com/magento/magento2/commit/bfce0265e495771e90350d206dbb0bf21132e879)
 - [472c1415b2697aef667402a9493a0db2f8601ec4](https://github.com/magento/magento2/commit/472c1415b2697aef667402a9493a0db2f8601ec4)

The fix will be available with the upcoming 2.3.0 release.

**Comment by magento-engcom-team** [comment]
Hi @dnadle. Thank you for your report. 
The issue has been fixed in magento/magento2#16564 by @dankhrapiyush in 2.2-develop branch
Related commit(s):
 - [873471a3064c831af1b2eb22e56af939e44e97a0](https://github.com/magento/magento2/commit/873471a3064c831af1b2eb22e56af939e44e97a0)
 - [9d84b2cc8a2cf2de7506b9809b035ed0077ac0eb](https://github.com/magento/magento2/commit/9d84b2cc8a2cf2de7506b9809b035ed0077ac0eb)
 - [d495275f4b65d16dbd6fda682b2c94e7cf4e6a02](https://github.com/magento/magento2/commit/d495275f4b65d16dbd6fda682b2c94e7cf4e6a02)
 - [b827bd19118c1de262d99c1ea9fd3f3a22c51872](https://github.com/magento/magento2/commit/b827bd19118c1de262d99c1ea9fd3f3a22c51872)

The fix will be available with the upcoming 2.2.6 release.

**Comment by sidolov** [comment]
Hi @dnadle. Thank you for your report. 
The issue has been fixed in magento/magento2#16640 by @gelanivishal in 2.3-develop branch
Related commit(s):
 - [6fd1eda49bb1da193d53eab7f0fe9dae1a04d605](https://github.com/magento/magento2/commit/6fd1eda49bb1da193d53eab7f0fe9dae1a04d605)
 - [f00c37a91d155f2cab74d747a5f819c3b408dede](https://github.com/magento/magento2/commit/f00c37a91d155f2cab74d747a5f819c3b408dede)
 - [1888b1eac681c21ba1f40642f65fe4aa65b3b7b1](https://github.com/magento/magento2/commit/1888b1eac681c21ba1f40642f65fe4aa65b3b7b1)
 - [0653f0946b676a29b7c2ee924cde6aa5d5f3ea20](https://github.com/magento/magento2/commit/0653f0946b676a29b7c2ee924cde6aa5d5f3ea20)

The fix will be available with the upcoming 2.3.0 release.

**Comment by VladimirZaets** [comment]
Hi @dnadle. Thank you for your report. 
The issue has been fixed in magento/magento2#16595 by @gelanivishal in 2.2-develop branch
Related commit(s):
 - [1d9cf03eab23766d25bfec2d10f9a312b9176f83](https://github.com/magento/magento2/commit/1d9cf03eab23766d25bfec2d10f9a312b9176f83)
 - [475cf0eed09c7a1be1de23bebca24ed96d310e03](https://github.com/magento/magento2/commit/475cf0eed09c7a1be1de23bebca24ed96d310e03)
 - [0f57a3076329fc72c372972c432c7ef41685cb06](https://github.com/magento/magento2/commit/0f57a3076329fc72c372972c432c7ef41685cb06)
 - [af58cb6d2132e1f85e048df54297d9621389a642](https://github.com/magento/magento2/commit/af58cb6d2132e1f85e048df54297d9621389a642)

The fix will be available with the upcoming 2.2.6 release.

**Comment by VladimirZaets** [comment]
Hi @dnadle. Thank you for your report. 
The issue has been fixed in magento/magento2#16986 by @gelanivishal in 2.1-develop branch
Related commit(s):
 - [2e4bcf13fb573f793298b483c7f6f769af9acbdf](https://github.com/magento/magento2/commit/2e4bcf13fb573f793298b483c7f6f769af9acbdf)

The fix will be available with the upcoming 2.1.15 release.

**Comment by magento-engcom-team** [comment]
Hi @dnadle. Thank you for your report. 
The issue has been fixed in magento/magento2#16956 by @gelanivishal in 2.3-develop branch
Related commit(s):
 - [f3502dafbeb3b525ab12ad6c0045efac8274951b](https://github.com/magento/magento2/commit/f3502dafbeb3b525ab12ad6c0045efac8274951b)
 - [9d15eccb97937cee8fdca373010233b7397be302](https://github.com/magento/magento2/commit/9d15eccb97937cee8fdca373010233b7397be302)
 - [6897f275e9e8176aeb2c11fb7f3f695bbc94d6e5](https://github.com/magento/magento2/commit/6897f275e9e8176aeb2c11fb7f3f695bbc94d6e5)

The fix will be available with the upcoming 2.3.0 release.
